packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -439,23 +454,67 @@ export class PipelineEditor extends React.Component<
 
           // prepare pipeline submission details
           const pipelineFlow = this.canvasController.getPipelineFlow();
-          pipelineFlow.pipelines[0]['app_data']['title'] =
-            result.value.pipeline_name;
-          // TODO: Be more flexible and remove hardcoded runtime type
+          const pipeline_path = this.widgetContext.path;
+
+          const pipeline_dir = Path.dirname(pipeline_path);
+          const pipeline_name = Path.filename(pipeline_path);
+          const pipeline_export_format = result.value.pipeline_filetype;
+          const pipeline_export_path =
+            pipeline_dir + pipeline_name + '.' + pipeline_export_format;
```

Comment:
if you update to use node's path lib this line would also need to change some since that library handles filename differently.
```suggestion
          const pipeline_dir = path.dirname(pipeline_path);
          const pipeline_name = path.basename(pipeline_path, path.extname(pipeline_path));
          const pipeline_export_format = result.value.pipeline_filetype;
          const pipeline_export_path =
            pipeline_dir + '/' + pipeline_name + '.' + pipeline_export_format;
```